lib/collection/src/operations/universal_query/collection_query.rs

```diff
@@ -46,6 +47,15 @@ impl CollectionQueryRequest {
     pub const DEFAULT_WITH_PAYLOAD: WithPayloadInterface = WithPayloadInterface::Bool(false);
 }
 
+/// Lightweight representation of a query request to implement the [RetrieveRequest] trait.
+#[derive(Debug)]
+pub struct CollectionQueryResolveRequest {
+    pub vector_query: VectorQuery<VectorInput>,
```

Comment:
nice

Reply:
Is it possible to make this a reference so that we don't have to clone raw vectors?